ir-gen-cli-bundler/conjureircli/run.go

```diff
@@ -73,37 +73,57 @@ func Run(inPath, outPath string) error {
 	if err != nil {
 		return err
 	}
+	if err := ensureCLIExists(cliPath); err != nil {
+		return err
+	}
 	cmd := exec.Command(cliPath, "compile", inPath, outPath)
 	if output, err := cmd.CombinedOutput(); err != nil {
 		return errors.Wrapf(err, "failed to execute %v\nOutput:\n%s", cmd.Args, string(output))
 	}
 	return nil
 }
 
+func cliUnpackDir() string {
+	return path.Join(os.TempDir(), "_conjureircli")
+}
+
+func cliArchiveDir() string {
+	return path.Join(cliUnpackDir(), fmt.Sprintf("conjure-%v", conjureircli_internal.Version))
+}
+
 func cliCmdPath() (string, error) {
-	cacheDirPath := path.Join(os.TempDir(), "_conjureircli")
-	dstPath := path.Join(cacheDirPath, fmt.Sprintf("conjure-%v", conjureircli_internal.Version))
-	if err := ensureCLIExists(dstPath); err != nil {
-		return "", err
-	}
 	switch runtime.GOOS {
 	case "darwin", "linux":
-		return path.Join(dstPath, "bin", "conjure"), nil
+		return path.Join(cliArchiveDir(), "bin", "conjure"), nil
 	default:
 		return "", errors.Errorf("OS %s not supported", runtime.GOOS)
 	}
 }
 
-func ensureCLIExists(dstPath string) error {
-	if fi, err := os.Stat(dstPath); err == nil && fi.IsDir() {
+func ensureCLIExists(cliPath string) error {
+	if fi, err := os.Stat(cliPath); err == nil && fi.Mode().IsRegular() && fi.Size() > 0 {
 		// destination already exists
 		return nil
 	}
 
+	// destination does not exist or is malformed, remove the dest dir just in case of a previous bad install
```

Comment:
There is mismatched naming here and below between `destination dir` and `archive dir`

Reply:
fixed! Also added some comments to each directory so it's obvious what they are